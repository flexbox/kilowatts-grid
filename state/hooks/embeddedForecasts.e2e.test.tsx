import React from "react";
import { cleanup, renderHook, waitFor } from "@testing-library/react-native";
import { useEmbeddedForecasts } from "./embeddedForecasts";
import { Provider } from "react-redux";
import { createStore } from "../reducer";

const wrapper = (props: any) => {
  return <Provider store={createStore()} children={props.children} />;
};

describe("useEmbeddedForecasts/useEmbeddedForecasts", () => {
  test("can get live data response", async () => {
    const now = new Date("2024-01-25T12:15:00Z");
    const r = renderHook(() => useEmbeddedForecasts(now), { wrapper });
    await waitFor(() => {
      expect(r.result.current.data).toBeDefined();
    });
    console.log(r.result.current.data);
    r.unmount();
  }, 5000);

  afterEach(async () => {
    await cleanup();
  });
});
