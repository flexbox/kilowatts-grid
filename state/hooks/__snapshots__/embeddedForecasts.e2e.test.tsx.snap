// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useEmbeddedForecasts/useEmbeddedForecasts can get live data response 1`] = `null`;
